docs/platform-specification.md

```diff
@@ -309,6 +309,59 @@ Sketch uses 924 bytes (2%) of program storage space. Maximum is 32256 bytes.
 Global variables use 9 bytes (0%) of dynamic memory, leaving 2039 bytes for local variables. Maximum is 2048 bytes.
 ```
 
+#### Recipes to compute binary sketch size for more complex systems (since Arduino CLI >=0.21.0)
+
+A platform may provide a tool for the specific purpose to analize the binaries and compute the sketch size and memory
+usage statistics. This is especially useful for boards with non-trivial memory layouts where the classic reg-exp based
+approach is not sufficient.
+
+The command line to run is specified with the recipe **recipe.advanced_size.pattern**.
+
+The expected output from the tool is a JSON object with the following format:
+
+```json
+{
+  "output": "Your sketch use 2200 bytes of program memory out of 8192 (22%)\nThe static RAM used is 200 bytes (of 2048 max)\n",
+  "severity": "info",
+  "sections": [
+    { "name": "text", "size": 2200, "maxsize": 8192 },
+    { "name": "data", "size": 200, "maxsize": 2048 }
+  ]
+}
+```
+
+The meaning of the fields is the following:
+
+- `output`: is a preformatted text that is displayed as-is in console.
+- `severity`: indicates the warning level of the output messages, it must be `info`, `warning` or `error`. Warnings and
+  errors are displayed in red (or in a different color than normal output). Errors will make the build/upload fail.
+- `sections`: is an array containing the memory sections and their usage level. Each item represents a memory section.
+  This array is used to report memory usage in a machine-readable format if requested by the user.
```

Comment:
```suggestion
- `sections`: is an array containing the memory sections and their usage level. This array is used to report memory
  usage in a machine-readable format if requested by the user. Each item represents a memory section and may contain the
  following fields
  - `name`: an identifier for the section
  - `size`: the sketch size for the section
  - `maxsize`: the maximum size for the section
```
I feel like it is better to explicitly specify these fields rather than expecting the user to differentiate the arbitrary parts of the samples from the specification parts.

I see that Arduino CLI doesn't require any of these fields, so I don't know whether it makes sense to mark any as required in the specification.

Reply:
The fields are optional, so it's good as you suggested.